### ISSUE TITLE
Harden nullability checks in `Endpoint.SignatureEquals`

### ISSUE THREAD

**Issue by captainsafia** [dotnet/aspnetcore#52492]
Logs indicate that unhandled NREs are thrown from the `Endpoint.SignatureEquals` from the `RequestDelegateGenerator` under certain and rare occasions. Although we've not be able to identify what the impacted syntax is, we should try to harden the nullability checks here to avoid situations where they do happen.

**Comment by david-acker** [comment]
Out of curiosity, are there any examples of these logs available?

**Comment by captainsafia** [comment]
> Out of curiosity, are there any examples of these logs available?

Unfortunately not. 😅 

The logs don't provide insight into what kind of syntax triggered the NRE, just that it happens repeatedly on every generator run. I haven't been able to reproduce it outside the context of the logs.

**Comment by adityamandaleeka** [comment]
@captainsafia Is this still an issue?

**Comment by captainsafia** [comment]
@adityamandaleeka Yes, it's still an issue, but I would stick it in the backlog for now.
